### PR TITLE
CommitlogArchiver only has granularity to seconds for restore_point_in_time for CASSANDRA-19448

### PR DESCRIPTION
**Pull request by Maxwell-Guo**

### CODE REVIEW

File: test/data/jmxdump/cassandra-3.0-jmx.yaml [hunk 1]
```diff
@@ -1850,7 +1850,6 @@ org.apache.cassandra.db:type=Commitlog:
   - {access: read-only, name: RestoreCommand, type: java.lang.String}
   - {access: read-only, name: RestoreDirectories, type: java.lang.String}
   - {access: read-only, name: RestorePointInTime, type: long}
-  - {access: read-only, name: RestorePrecision, type: java.lang.String}
```
Comment (smiklosovic) [inline test/data/jmxdump/cassandra-3.0-jmx.yaml]: why is this here? we put precision back, no? applies for other versions for this file as well.

File: test/unit/org/apache/cassandra/db/commitlog/CommitLogArchiverTest.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                                 "restore_point_in_time", rpiTime));
+        CommitLogArchiver commitLogArchiver = CommitLogArchiver.getArchiverFromProperties(properties);
+        // set the archiver at the very beginning
+        commitLog.setCommitlogArchiver(commitLogArchiver);
+    }
+
+    @Test
+    public void testArchiver()
+    {
+        File dir = new File(dirName);
```
Comment (smiklosovic) [inline test/unit/org/apache/cassandra/db/commitlog/CommitLogArchiverTest.java]: @Maxwell-Guo  you should move this logic to `@Before`, that way, each test will have that directory empty (even it is not needed for it for now). It is just better practice.

File: src/java/org/apache/cassandra/db/commitlog/CommitLogArchiver.java [hunk 3]
```diff
@@ -311,4 +344,64 @@ private void exec(String command) throws IOException
...
+    }
+
+    /**
+     * compare the timestamps which are all microseconds with different precision
+     * @param left left microseconds
+     * @param right right microseconds
+     * @param precision the precision
+     * @return true if left is bigger than right
+     */
+    public boolean compareMicrosecondsWithPrecision(long left, long right, TimeUnit precision)
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/db/commitlog/CommitLogArchiver.java]: @Maxwell-Guo 

Please remove this whole method and put it back as it was when you merged my pull request. This is not necessary. Practically speaking, there will ever be the resolution in PartitionUpdate either microseconds or milliseconds (if even that). Nobody is going to put timestamps on e.g. day resolution ...

File: test/data/jmxdump/cassandra-3.0-jmx.yaml (hunk 1 above)
Comment (Maxwell-Guo) [reply test/data/jmxdump/cassandra-3.0-jmx.yaml]: yes, you are right. I forgot to restore this.
